### ISSUE TITLE
Tokenizers: Character encoding inconsistencies between __call__ and .convert_tokens_to_ids

### ISSUE THREAD

**Issue by FrancescoManfredi** [huggingface/transformers#31438]
### System Info

- `transformers` version: 4.41.2
- Platform: Linux-6.1.0-21-amd64-x86_64-with-glibc2.36
- Python version: 3.11.2
- Huggingface_hub version: 0.23.2
- Safetensors version: 0.4.3
- Accelerate version: 0.31.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.3.0+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: YES
- Using distributed or parallel set-up in script?: NO

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```python
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("swap-uniba/LLaMAntino-3-ANITA-8B-Inst-DPO-ITA")
original_tokens = ["à", "è", "ì", "ò", "ù"]

# converting tokens to ids with __call__ gives a wrong encoding
ids_direct_call = tokenizer(original_tokens)['input_ids']
print(ids_direct_call)
# > [[6496], [4558], [24439], [22980], [15273]]

ids_direct_call = [i[0] for i in ids_direct_call]
print(ids_direct_call)
# > [6496, 4558, 24439, 22980, 15273]

tokens_direct_call = tokenizer.convert_ids_to_tokens(ids_direct_call)
print(tokens_direct_call)
# > ['Ãł', 'Ã¨', 'Ã¬', 'Ã²', 'Ã¹']

# Converting the original tokens with .convert_tokens_to_ids gives the
# correct encoding
ids_explicit_convert = tokenizer.convert_tokens_to_ids(original_tokens)
print(ids_explicit_convert)
# > [156, 164, 168, 174, 181]

tokens_explicit_convert = tokenizer.convert_ids_to_tokens(ids_explicit_convert)
print(tokens_explicit_convert)
# > ['à', 'è', 'ì', 'ò', 'ù']

```

### Expected behavior

I'd expect both methods to behave in the same way and maybe some straightforward way to explicitly suggest the correct encoding to use.  
  
Also I'd like to know if I'm missing something and this is actually an expected behavior.

**Comment by ArthurZucker** [comment]
Hey! I am not sure I understand why you would expect both methods to behave the same way? 
One is the full `encode`, which applies the `pre_tokenizer` , the `normalizer` while the other one, only converts the id to the corresponding token in the vocab. 

These methods are inherently different.

**Comment by FrancescoManfredi** [comment]
Right. I don't know what made me expect these two to perform the same action but now I understand the whole process better. Thank you for your answer and your work!
